tests/config/format-test.js

```diff
@@ -334,6 +327,11 @@ function runTest({
     return;
   }
 
+  // [prettierx]: disable test files by parser/options
+  if (isDisabled(filename, parser, formatOptions)) {
```

Comment:
This change is good because it means we don't avoid testing the `babel-ts` tests when they don't run into the bug we discovered, but it also means that `isDisabled` is a little less clear. I imagine a scenario in which someone adds another test to the disabled test list, assuming that they will be disabled in their entirety, even though that isn't the behaviour.

I've pushed up a small change that, I think, makes this a little clearer without making any overly complicated additions or changes. I also threw in some doc comments while I was at it.